### PR TITLE
Feature fullres abi tc

### PR DESCRIPTION
**Pull request by davidh-ssec**
This changes the ABI true color composites to have a full resolution version named `true_color` and a lower 2km resolution version at `true_color_2km`.

### CODE REVIEW

**Comment by davidh-ssec** [comment]
Also this includes fixes for the base reader and newer versions of six. I don't *think* it should be an issue for older versions, but I'm not 100% sure.

File: satpy/etc/composites/abi.yaml [hunk 1]
```diff
@@ -21,10 +25,35 @@ modifiers:
...
+    atmosphere: midlatitude summer
+    aerosol_type: marine_tropical_aerosol
+    prerequisites:
+    - name: C01
+      modifiers: [expander2, effective_solar_pathlength_corrected]
+    optional_prerequisites:
+    - satellite_azimuth_angle
+    - satellite_zenith_angle
+    - solar_azimuth_angle
+    - solar_zenith_angle
```
Comment (mraspaud) [inline satpy/etc/composites/abi.yaml]: I'm not convinced here. Why not just correct the channel and then expand it ?

File: satpy/composites/abi.py [hunk 2]
```diff
@@ -42,4 +48,29 @@ def __call__(self, projectables, **info):
...
+        g = (b + r) / 2 * 0.93 + 0.07 * c03_high
+
+        low_res_red = four_element_average(r)
+        low_res_red = np.repeat(np.repeat(low_res_red, 2, axis=0), 2, axis=1)
+        ratio = r / low_res_red
+
+        # make sure metadata is copied over
+        # copy red channel area to get correct resolution
+        g *= ratio
+        g.info = c03.info.copy()
```
Comment (mraspaud) [inline satpy/composites/abi.py]: Maybe remove the wavelength information from the info ?

**Comment by davidh-ssec** [comment]
As discussed on slack the green band metadata is ignored if it is different from the other channels when the RGB is created and metadata merged together. So changing wavelength or name has no affect and is not included anywhere.

For the argument of when expanding should take place it isn't guaranteed that it isn't a huge waste of memory. Meaning if you expand the lower resolution channels, then do atmospheric correction, you are probably going to use 4-8x more memory than you would if you expanded it after the correction.

**Pull request merged**